component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java

```diff
@@ -286,6 +292,17 @@ private X509Certificate parseCertificate(String content) throws CertificateExcep
                 .generateCertificate(new ByteArrayInputStream(decoded));
     }
 
+    /**
+     * Check if the string contains percent-encoded sequences.
+     *
+     * @param str input string value.
+     * @return returns true if string is url encoded.
+     */
+    private boolean isUrlEncoded(String str) {
+
+        return str.matches(".*(%[0-9a-fA-F]{2}).*");
```

Comment:
Is it reliable to determine whether a certificate is UTF-8 encoded or not using a regex ? Don't we have any libraries to check this?

Reply:
Couldn't find any

Reply:
Anyway relying on the regex is not reliable and we can not guarantee on the functionality. We may need to look for an alternative.

Reply:
Will try to base64 decode the cert first (as in the earlier implementation) and if it throws an exception will urlDecode before base64 decoding

Reply:
Changed with https://github.com/wso2-extensions/identity-oauth-addons/pull/132/commits/88128bd2abe38aa31609ed5f3eb535b774b50a06